nx-dev/ui-markdoc/src/lib/tags/youtube.component.tsx

```diff
@@ -0,0 +1,46 @@
+import { Schema } from '@markdoc/markdoc';
+
+export const youtube: Schema = {
+  render: 'YouTube',
+  attributes: {
+    src: {
+      type: 'String',
+      required: true,
+    },
+    title: {
+      type: 'String',
+      required: true,
+    },
+    width: {
+      type: 'String',
+      default: '50%',
+    },
+    caption: {
+      // Added caption attribute here
+      type: 'String',
+      required: false, // Not required since it's optional
+    },
+  },
+};
+
+export function YouTube(props: any): JSX.Element {
+  return (
+    <div className="text-center">
+      {' '}
+      {/* Center alignment applied to the container */}
+      <iframe
+        {...props}
```

Comment:
We have just 4 props and we explicitly set 2 of them already. Let's make the remaining two explicit as well